### ISSUE TITLE
CA2002 should verify also usages of Monitor.Enter and Monitor.TryEnter

### ISSUE THREAD

**Issue by cezarypiatek** [dotnet/roslyn-analyzers#2744]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

2.9.4

#### Diagnostic ID

[CA2002](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca2002-do-not-lock-on-objects-with-weak-identity.md)

#### Repro steps

```csharp
Monitor.Enter("SampleString")
```
#### Expected behavior
Usages of `Monitor.Enter` and `Monitor.TryEnter` should be also verified against `CA2002` because they work in the same way as `lock()` statement - `lock()` is a syntax sugar that is compliled to 

```csharp
try{
 Monitor.Enter();
}
finally
{
 Monitor.Exit();
}
```

#### Actual behavior
Only usages of `lock()` statement are verified against `CA2002`

I can prepare a PR with that fix but I would like to get confirmation of my expectation against `CA2002`  before I start any work.

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-analysis for views on the proposed rule enhancement

**Comment by sharwell** [comment]
Looks good to me. Note that we only need to look at `Monitor.Enter` and `Monitor.TryEnter`.

**Comment by cezarypiatek** [comment]
Ok, thanks for the confirmation. I'll try to implement it over the weekend
